src/AppInstallerCommonCore/FileLogger.cpp

```diff
@@ -134,4 +124,22 @@ namespace AppInstaller::Logging
                 catch (...) {}
             }).detach();
     }
+
+    void FileLogger::OpenFileLoggerStream() 
+    {
+        // Prevent inheritance to ensure log file handle is not opened by other processes.
+        FILE* filePtr;
+        if (!fopen_s(&filePtr, m_filePath.string().c_str(), "w") && filePtr != NULL)

```

Comment:
```suggestion
        errno_t fopenError = _wfopen_s(&filePtr, m_filePath.wstring().c_str(), L"w");
```

Use wide strings and capture the return value from `_wfopen_s` rather than using the global `errno`.  If you want to guard against a null pointer even when the return value indicates success, do that inside the block.